### ISSUE TITLE
Terminus Station Component

### ISSUE THREAD

**Issue by peterlai1** [transit-analytics-lab/spur#54]
Proposal and specification for new terminus station component.

### Description

A terminus station with multiple tracks that can also serve as a yard, where trains could spawn and disappear.

### Component Parameters

- Number of tracks with platform
- Overall capacity
- Boarding / alighting durations or volume parameters 

### Logic

- Essentially a merged version of SimpleYard and Multi-Track Station
- Number of storage-only tracks = Overall capacity - Number of tracks with platform
- If a train needs to board / alight, assign a track with platform, and apply dwell time using distribution as in the Multi-Track Station component
- Trains do not 'bypass' the station; they are immediately made available and it's up to the tour configuration to decide how long they stay there for using departure and arrival time arguments

### Future Extensions

- Assignment of preferred track

**Comment by wklumpen** [comment]
Hey @peterlai1 did we end up adding this one into the model?

**Comment by peterlai1** [comment]
> Hey @peterlai1 did we end up adding this one into the model?

No, we didn't end up having the time to create a new standalone component for this in the thesis project scope - we just used `MultiTrackStation` instead for terminus stations as well. That said, it would still be worth creating!

**Comment by wklumpen** [comment]
Okay I'll add this to "future" then!
